homework/index.js

```diff
@@ -30,15 +30,106 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
+  const root = document.getElementById('root');
+  function fetchAllRepoData(url) {
+    return fetch(url)
+      .then(res => res.json())
+      .then(data => {
+        const navDiv = createAndAppend('div', root, { class: 'nav' });
+        const navHeader = createAndAppend('div', navDiv, { class: 'nav-header' });
+        const navTitle = createAndAppend('div', navHeader, { class: 'nav-title' });
+        navTitle.innerText = 'HYF Repositories';
+        const selectEl = createAndAppend('select', navDiv, { id: 'getRepoData' });
+        data.sort((a, b) => (a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 1));
+        data.forEach(res => {
+          createAndAppend('option', selectEl, { text: res.name, value: res.id });
+        });
+      })
+      .catch(err => {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
-    });
+      });
+  }
+
+  function getRepoData(url) {
+    const selectRepo = document.getElementById('getRepoData');
+    const container = createAndAppend('div', root, { class: 'container' });
+    function fetchSingleData() {
+      return fetch(url)
+        .then(res => res.json())
+        .then(repData => {
+          container.innerHTML = '';
+          const filteredData = repData.filter(el => el.id === Number(selectRepo.value));
```

Comment:
I think you try to re-fetch the repos data so that you can apply this filter operation to get data of the selected repo(which is the first repo by default at page load). This can be avoided by simply accessing the 0th element of the repositories array as it represents the first element of an array. Thus, when you are creating the repositories container elements, you can use the 0th index instead of applying a filter operation first. This would allow you to do away with the second fetch operation. You might have to use a parameter to this function to get access to the repos data. 😉